hooks/workfiles_management.py

```diff
@@ -85,7 +86,10 @@ def register_workfile(self, name, version, context, work_template, path, descrip
             "sg_path": sg_path,
             "project": context.project
         }
-
+        if "name" in work_template.keys:
+            new_workfile["sg_name_field"] = work_template.get_fields(path)["name"]
```

Comment:
what is the difference between `work_template.get_fields(path)["name"]` and the input `name` parameter?